goplay/src/test/java/be/tapped/goplay/e2e/ProgramRepoTest.kt

```diff
@@ -0,0 +1,57 @@
+package be.tapped.goplay.e2e
+
+import arrow.fx.coroutines.parTraverse
+import be.tapped.goplay.content.AllProgramsHtmlJsonExtractor
+import be.tapped.goplay.content.HttpProgramRepo
+import be.tapped.goplay.content.ProgramDetailHtmlJsonExtractor
+import be.tapped.goplay.epg.GoPlayBrand
+import be.tapped.goplay.httpClient
+import be.tapped.goplay.jsonSerializer
+import io.kotest.assertions.arrow.core.shouldBeRight
+import io.kotest.core.spec.style.FreeSpec
+
+internal class ProgramRepoTest : FreeSpec({
```

Comment:
E2E test that hits the real https://goplay.be site.  

Loops over every 300 of the available programs and retrieve all the associated metadata both by id and link.  

Retrieves the most popular programs by brand

Reply:
562 tests